orttraining/orttraining/core/framework/gradient_graph_builder.cc

```diff
@@ -55,9 +55,10 @@ GradientGraphBuilder::GradientGraphBuilder(Graph* graph,
 
     const Node* node = graph_->GetProducerNode(name);
     if (!node) {
-      ORT_THROW(name, " couldn't find the producer node.");
+      LOGS(logger_, WARNING) << name << " couldn't find the producer node." << std::endl;
+    } else {
+      y_nodes_.insert(node);
```

Comment:
@SherlockNoMad @suffiank Do you guys see an issue with that approach?

Reply:
I'm not sure I understand what's going on here. If we ignore the `y` node on L58, do we still end up building the correct backward graph for the `x -> x` case? The other potential issue would be that the method now *won't* reject invalid graphs where the producer doesn't exist - although perhaps that's not a big issue if we have validation at other layers.

Perhaps it would make sense to tighten the condition? e.g. Could we also test whether the "producer" for the node is one of the graph inputs (or one of the `x_node_arg_names`)? If we did that, we wouldn't have to emit a warning, and could still `ORT_THROW` on truly invalid graphs.

Reply:
@SherlockNoMad  @iK1D do you guys have any comment on that?

Reply:
@codemzs maybe you know this too :)

Reply:
@spandantiwari

Reply:
I don't understand the issue here. 
y_node_arg_names should be a valid set of names, as the starting point for backward propagation?

When would we run into the case where y_node is not valid?

Reply:
```
def test_model_without_trainable_parameters(device):

    model = torch.nn.ReLU()
    model = ORTModule(model).to(device)

    batch_size = 5
    nb_classes = 2
    in_features = 10
    input = torch.randn(batch_size, in_features, device=device)
    target = torch.empty(batch_size, dtype=torch.long, device=device).random_(nb_classes)
    _ = model(input)
```

Reply:
That generates a graph that is 

x -> x
(input) -> (output)